### PR TITLE
feat: add scaffold impl of EthFilter API handler

### LINKED ISSUE #1247: Add scaffold impl of EthFilter API handler
### Describe the feature

Similar to #982 , just scaffolding, no impls yet

## TODO
Add handler structs and impl corresponding server trait for `EthFilter`
* EthFilter

similar to:

https://github.com/paradigmxyz/reth/blob/9470943babf8afe98a10db729954a374470afcb7/crates/rpc/rpc/src/eth/pubsub.rs#L10-L17

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leruaa**
Closes #1247

### CODE REVIEW

File: crates/rpc/rpc/src/eth/filter.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
+use std::{future::Future, pin::Pin, sync::Arc};
+
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/filter.rs:2]:
```suggestion
```

File: crates/rpc/rpc/src/eth/filter.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        'life0: 'async_trait,
+        Self: 'async_trait,
+    {
+        todo!()
+    }
+
+    fn filter_logs<'life0, 'async_trait>(
+        &'life0 self,
+        _index: Index,
+    ) -> Pin<Box<dyn Future<Output = RpcResult<Vec<Log>>> + Send + 'async_trait>>
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/filter.rs:56]: this is the output `async_trait` generics, you can use the same `async` function the trait uses if you annotate this trait impl with #[async_trait] as well.

File: crates/rpc/rpc/src/eth/filter.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    fn uninstall_filter(&self, _index: Index) -> RpcResult<bool> {
+        todo!()
+    }
+
+    fn logs<'life0, 'async_trait>(
+        &'life0 self,
+        _filter: Filter,
+    ) -> Pin<Box<dyn Future<Output = RpcResult<Vec<Log>>> + Send + 'async_trait>>
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/filter.rs:71]: same

**Review by mattsse — CHANGES_REQUESTED** [review]
tysm,

1 nit re async trait

**Comment by mattsse** [comment]
once merged you can open a follow up that install the missing eth handlers if the eth module is selected.

I'll open an issue and tag you.

**Review by mattsse — APPROVED** [review]
lgtm

**Comment by leruaa** [comment]
Wooow what happened here ?! The PR hasn't been merged ?

**Pull request closed without merging**